binding/web/src/rhino.ts

```diff
@@ -0,0 +1,843 @@
+    this._slotsAddressAddressAddress = handleWasm.slotsAddressAddressAddress;
+    this._valuesAddressAddressAddress = handleWasm.valuesAddressAddressAddress;
+
+    this._memoryBuffer = new Int16Array(handleWasm.memory.buffer);
+    this._memoryBufferUint8 = new Uint8Array(handleWasm.memory.buffer);
+    this._memoryBufferView = new DataView(handleWasm.memory.buffer);
+    this._processMutex = new Mutex();
+  }
+
+  /**
+   * Get Rhino engine version.
+   */
+  get version(): string {
+    return Rhino._version;
+  }
+
+  /**
+   * Get frame length.
+   */
+  get frameLength(): number {
+    return Rhino._frameLength;
+  }
+
+  /**
+   * Get sample rate.
+   */
+  get sampleRate(): number {
+    return Rhino._sampleRate;
+  }
+
+  /**
+   * Get context info.
+   */
+  get contextInfo(): string {
+    return Rhino._contextInfo;
+  }
+
+  /**
+   * Creates an instance of the Rhino speech-to-intent engine using a base64'd string
+   * of the model file. The model size is large, hence it will try to use the
+   * existing one if it exists, otherwise saves the model in storage.
+   *
+   * @param accessKey AccessKey generated by Picovoice Console.
+   * @param context RhinoContext object containing a base64
+   * representation of or path to public binary of a Rhino context model .
+   * @param modelBase64 The model in base64 string to initialize Rhino.
+   * @param options Optional configuration arguments.
+   * @param options.sensitivity Inference sensitivity. It should be a number within [0, 1].
+   * A higher sensitivity value results in fewer misses at the cost of (potentially)
+   * increasing the erroneous inference rate.
+   * @param options.endpointDurationSec Endpoint duration in seconds.
+   * An endpoint is a chunk of silence at the end of an utterance that marks
+   * the end of spoken command. It should be a positive number within [0.5, 5].
+   * A lower endpoint duration reduces delay and improves responsiveness. A higher endpoint duration
+   * assures Rhino doesn't return inference pre-emptively in case the user pauses before finishing the request.
+   * @param options.requireEndpoint If set to `true`, Rhino requires an endpoint (a chunk of silence)
+   * after the spoken command. If set to `false`, Rhino tries to detect silence, but if it cannot,
+   * it still will provide inference regardless. Set to `false` only if operating in an
+   * environment with overlapping speech (e.g. people talking in the background).
+   * @param options.customWritePath Custom path to save the model in storage.
+   * Set to a different name to use multiple models across `rhino` instances.
+   * @param options.forceWrite Flag to overwrite the model in storage even if it exists.
+   * @param options.version Rhino model version. Set to a higher number to update the model file.
+   * @returns An instance of the Rhino engine.
+   */
+  public static async fromBase64(
+    accessKey: string,
+    context: RhinoContext,
```

Comment:
so I assume you are going to add the inferenceCallback here, right?

Reply:
Yup :)